lib/execution-info.js

```diff
@@ -0,0 +1,526 @@
+'use strict';
+
+const utils = require('./utils');
+const types = require('./types');
+const errors = require('./errors');
+
+/**
+ * A base class that represents a wrapper around the user provided query options with getter methods and proper
+ * default values.
+ */
+class ExecutionInfo {
+
+  /**
+   * Creates a new instance of {@link ExecutionInfo}.
+   */
+  constructor() {
+  }
+
+  /**
+   * Creates an empty instance, where all methods return undefined, used internally.
+   * @ignore
+   * @return {ExecutionInfo}
+   */
+  static empty() {
+    return new ExecutionInfo();
+  }
+
+  /**
+   * Determines if the stack trace before the query execution should be maintained.
+   * @abstract
+   * @returns {Boolean}
+   */
+  getCaptureStackTrace() {
+
+  }
+
+  /**
+   * Gets the [Consistency level]{@link module:types~consistencies} to be used for the execution.
+   * @abstract
+   * @returns {Number}
+   */
+  getConsistency() {
+
+  }
+
+  /**
+   * Key-value payload to be passed to the server. On the server side, implementations of QueryHandler can use
+   * this data.
+   * @abstract
+   * @returns {Object}
+   */
+  getCustomPayload() {
+
+  }
+
+  /**
+   * Gets the amount of rows to retrieve per page.
+   * @abstract
+   * @returns {Number}
+   */
+  getFetchSize() {
+
+  }
+
+  /**
+   * When a fixed host is set on the query options and the query plan for the load-balancing policy is not used, it
+   * gets the host that should handle the query.
+   * @returns {Host}
+   */
+  getFixedHost() {
+
+  }
+
+  /**
+   * Gets the type hints for parameters given in the query, ordered as for the parameters.
+   * @abstract
+   * @returns {Array|Array<Array>}
+   */
+  getHints() {
+
+  }
+
+  /**
+   * Determines whether the driver must retrieve the following result pages automatically.
+   * <p>
+   *   This setting is only considered by the [Client#eachRow()]{@link Client#eachRow} method.
+   * </p>
+   * @abstract
+   * @returns {Boolean}
+   */
+  getIsAutoPage() {
```

Comment:
Not sure if there is a convention, but would it make sense for these boolean-based property getters to be simply `isAutoPage()`, `isBatchCounter`, `isBatchLogged`, etc.?

Reply:
I was going for `getX()` and `setX()` methods but maybe is not a good idea for booleans. Given that set methods are only internal, the user wont see this (get/set) convention, I'll change them to `isX()` 👍